### PR TITLE
Object Serialization and Attribute Embedding

### PR DESCRIPTION
**Pull request by faultyserver**
Dealing with #5 (finally) and #11.

Object Serialization takes place implicitly when the WAMP client in Transport (provided by [ruby_wamp_client](https://github.com/ericchapman/ruby_wamp_client)) calls `to_json` on args and kwargs.

Serialization is implemented as a module (`ObjectSerialization`) that `Shark::Object` includes. `Shark::Object` also now includes `Configurable`, and there are a number of options that can be changed to affect the serialization (see `config/serialization.rb` for details).

### CODE REVIEW

**Comment by faultyserver** [comment]
I'm ready to merge this, but I haven't tested it through crossbar yet, so I can't effectively analyze the output being generated (I don't have a staging instance set up).

Probably tomorrow (if it doesn't mess you up too much, @elliottwilliams) I'll pause production for a few minutes to look at how this is working.

File: middlewares/conductor/station_events.rb [hunk 1]
```diff
@@ -5,31 +5,31 @@ class Conductor
...
+  register_handler 'stations', :deactivate do |event|
     # No conducting is necessary
   end
 
   # depart -> [vehicle] {**defaults}
   #   once
   # Sent when a vehicle departs from this Station. `vehicle` will be the
   # attributes of the vehicle that departed.
-  register_handler 'stations', :depart do |channel, args, kwargs|
+  register_handler 'stations', :depart do |event|
```
Comment (elliottwilliams) [inline middlewares/conductor/station_events.rb:32]: Out of curiosity, what makes this empty handler necessary?

**Comment by elliottwilliams** [comment]
Looks excellent. Feel free to mess with prod. I'll probably be working on Proper tomorrow but can use a local instance.

The merge conflict, btw, was me correcting the same typo you did, so you can easily rebase and drop that commit.

File: middlewares/conductor/station_events.rb (hunk 1 above)
Comment (faultyserver) [inline middlewares/conductor/station_events.rb:32]: I added those just for consistency and clarity.

With these, Conductor has a handler implementation (granted, some empty ones) for every event on the "WAMP Events and RPCs" wiki page.

They could easily be left for `default_event_handler` to take care of, but adding all of them for completeness wasn't much extra trouble.

**Comment by faultyserver** [comment]
Everything seems to be working. I just need to set `nested_embedded_attributes` for `Shark::Vehicle`, and default `embed_associated_objects` to false instead of true.

Will merge after these changes.

**Pull request merged**